File: haskell/parhello.hs
#!/usr/bin/env runhaskell

-- Andrew Pennebaker

import Control.Concurrent.ParallelIO.Global
import System.IO

main = do
  -- By default, buffering prevents some IO nondeterminism
  hSetBuffering stdout NoBuffering

  parallel_ $ map putChar "Hello World!"

  stopGlobalPool

File: c/parhello.c
#include <omp.h>
#include <stdio.h>
#include <string.h>

int main() {
  char *s = "Hello World!\n";

  size_t i;

#pragma omp parallel for
  for (i = 0; i < strlen(s); i++) {
    putc(s[i], stdout);
  }

  return 0;
}
